hibernate-core/src/main/java/org/hibernate/cfg/annotations/CollectionBinder.java

```diff
@@ -663,14 +663,6 @@ else if ( comparatorSort != null ) {
 			}
 		}
 
-		if ( isSortedCollection ) {
-			if ( ! hadExplicitSort && !hadOrderBy ) {
-				throw new AnnotationException(
-						"A sorted collection must define and ordering or sorting : " + safeCollectionRole()
-				);
-			}
-		}
-
```

Comment:
I manually compared to explicit `@SortNatural` case and found everything is the same(only `hadExplicitSort` values are different but it is used internally in this method and its main usage is the above code snippet), so removing the above code snippet seems enough to achieve the goal of this PR, unless I were wrong.